### PR TITLE
Enum descriptions in new settings editor via Suggest widget

### LINKED ISSUE #55796: Show setting enumDescriptions inside new dropdown control
image

- The doc part is shown on hover, if that value has an enumDescription. Not all do, so maybe we show an (i) indicator on ones that do

Can we use the suggest widget, or is it easier to build on top of the menu control?

Accessibility.

### PR DESCRIPTION
**Pull request by JacksonKearl**
Closes #55796

This is a somewhat quick and dirty initial implementation. It seems pretty slow on Code - OSS but when I made a full minified build I didn't notice anything particularly slow. It also isn't properly keyboard navigable.

If we decide to go with this idea we'd remove the input box all-together and probably replace it with a button that triggers the dropdown, or just a div and wire up the onspace, onclick, onenter, tab stopping, etc. manually. 

It probably looks bad on Mac, as the rounded corners may not hide the input sufficiently, but I don't have anything with me to test.

moredetailed

### CODE REVIEW

**Comment by JacksonKearl** [comment]
I don't know of a good way to get markdown links to work as we don't have access to the markdown `actionProvider` for the suggest detail view. Other markdown is fine though.

**Comment by JacksonKearl** [comment]
Notes from sitdown:
- Look into if the description view should be always enabled
- Add `i` markers for items with a description view
- Scrolling is broken

**Comment by roblourens** [comment]
And 

- What will it take to show the description on hover?

I think this and the scrolling issue are blockers.

**Comment by roblourens** [comment]
"select an element when I hover it" is what I think would feel natural for mouse interaction.

**Comment by JacksonKearl** [comment]
Okay, so same sort of thing as menus. I wonder if that should be added to `Suggest`s as a whole.

**Comment by roblourens** [comment]
It's possible, although in normal usage I think people shouldn't be using the mouse.

**Comment by JacksonKearl** [comment]
For me the question is whether I sublclass the suggester for this or implement it for everywhere

**Comment by roblourens** [comment]
I think either subclass or put it behind an option in the control itself, whatever is most natural. I could see turning it on in the editor, since otherwise there is no way to see an enumDescription with only the mouse.

**Comment by roblourens** [comment]
Maybe @ramya-rao-a has an opinion since she has worked on the suggest widget.

**Comment by JacksonKearl** [comment]
Actually native behavior is to remove the dropdown on scroll so this might actually be the expected behavior.

File: src/vs/editor/contrib/suggest/suggestWidget.ts [hunk 1]
```diff
@@ -458,6 +458,7 @@ export class SuggestWidget implements IContentWidget, IVirtualDelegate<ICompleti
 			editor.onDidLayoutChange(() => this.onEditorLayoutChange()),
 			this.list.onSelectionChange(e => this.onListSelection(e)),
 			this.list.onFocusChange(e => this.onListFocus(e)),
+			this.list.onMouseOver(e => this.list.setFocus([e.index])),
```
Comment (JacksonKearl) [inline src/vs/editor/contrib/suggest/suggestWidget.ts]: @ramya-rao-a this adds focus on mouseover to all suggest widgets. This makes sense to me but what do you think? I could put it behind a flag somewhere.

**Comment by JacksonKearl** [comment]
Opting for the alternative implementation in #57050.

**Pull request closed without merging**